rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -291,7 +295,10 @@ impl PendingOperation for PendingMessage {
         {
             Ok(tx_cost_estimate) => tx_cost_estimate,
             Err(err) => {
-                return self.on_reprepare(Some(err), ReprepareReason::ErrorEstimatingGas);
+                let reason = self
+                    .clarify_reason(ReprepareReason::ErrorEstimatingGas)
+                    .await;
+                return self.on_reprepare(Some(err), reason);
```

Comment:
I think any time we call `on_reprepare` on this file we should consider parsing with the op-verifier, including in the submit step